### ISSUE TITLE
Cannot read property 'm_iPrimaryAmmoType' of undefined

### ISSUE THREAD

**Issue by thorebear** [saul/demofile#106]
I have noticed this error when parsing some demos recently:
```
Error during parsing: TypeError: Cannot read property 'm_iPrimaryAmmoType' of undefined
    at Weapon.getProp (...demofile/dist/entities/networkable.js:40:37)
```

Example of broken demos is map 2 (train) and 3 (inferno) from here: 
https://www.hltv.org/matches/2330438/faze-vs-big-eleague-csgo-invitational-2019

Using demofile version: 1.0.2.

**Comment by saul** [comment]
Please can you give the full stack trace and the code you’re using to parse the file? Thanks

**Comment by thorebear** [comment]
Full stack trace:
```
    at Weapon.getProp (../demofile/dist/entities/networkable.js:40:37)
    at Entities._readNewEntity (../demofile/dist/entities.js:348:37)
    at Entities._handlePacketEntities (../demofile/dist/entities.js:422:22)
    at DemoFile.demo.on.e (../demofile/dist/entities.js:97:49)
    at DemoFile.emit (events.js:188:13)
    at DemoFile._handleDemoPacket (../demofile/dist/demo.js:172:22)
    at DemoFile._parseRecurse (../demofile/dist/demo.js:220:26)
    at processImmediate (timers.js:632:19)
```

The error can be reproduced by the following code:
```
const demofile = require('demofile');
const fs = require('fs');

const demo_path = "faze-vs-big-m2-train.dem"

fs.readFile(demo_path, function(err, buffer) {
    const demo = new demofile.DemoFile();
    demo.gameEvents.emit = function() {
        var emitType = arguments['0'];
        var info = arguments['1'];
    }
    demo.on("end", e => {
        if (e.error) {
            console.error("Error during parsing:", e.error);
        } else {
            console.log("Success")
        }
    })


    demo.parse(buffer);
});
```

**Comment by CodeAndRun** [comment]
Have the same issue among about 15% of demos heap I'm trying to parse (about 2k demos).
My guess is a rare weapon?
If you need more broken demos please let me know.

**Comment by saul** [comment]
This has been fixed in v1.0.4
